pro_gan_pytorch/CustomLayers.py

```diff
@@ -419,8 +419,10 @@ def __init__(self, in_channels, num_classes, use_eql):
             self.conv_3 = Conv2d(in_channels, 1, (1, 1), bias=True)
 
         # we also need an embedding matrix for the label vectors
-        self.label_embedder = Embedding(num_classes, in_channels, max_norm=1)
-
+        self.label_embedder = Embedding(num_classes, in_channels)
+        norms = th.norm(self.label_embedder.weight, p=2, dim=1).data
+        self.label_embedder.weight.data = self.label_embedder.weight.data.div(norms.view(num_classes,1).
```

Comment:
@arvindmvepa, 
Could you please explain how this norm constraint is working here?
From what I understand, this normalization would be executed only once during this `__init__` right?
the `max_norm = 1` on the other hand has a hook which ensures that the embeddings' norm never exceeds the value 1.
Could you please check this?

Cheers :beers:!
@akanimax

Reply:
i got the solution from here: https://discuss.pytorch.org/t/normalizing-embeddings/7696/2. my understanding is that max_norm=1 ensures that the norm never exceeds 1 while this solution normalizes the weights regardless if they exceed 1 or not. i will double check this but i believe that it would continue to normalize during training, not just initially.

Reply:
I think you're right actually - based on some tests i've been doing for the last few hours it seems like the normalization term doesn't change with training, so it doesn't update as the embedding weights update. I tried removing `max_norm` altogether; while the discriminator loss is really high initially, as the model trains it reduces substantially, so that's another option.